addons/sale_purchase/models/sale_order_line.py

```diff
@@ -194,6 +194,7 @@ def _purchase_service_prepare_line_values(self, purchase_order, quantity=False):
             uom_id=self.product_id.uom_po_id
         )
 
+        analytic_distribution = self.analytic_distribution if self.analytic_distribution else {self.order_id.analytic_account_id.id: 100}
```

Comment:
We should also check if `self.order_id.analytic_account_id` is defined before setting the analytic_distribution with its id.

Also instead of setting False to `analytic_distribution`, we could change a little bit the method to only add `analytic_distribution` in the vals once it is needed.

Something like: 

```diff
diff --git a/addons/sale_purchase/models/sale_order_line.py b/addons/sale_purchase/models/sale_order_line.py
index 161cea9f3337..57d88867c2a8 100644
--- a/addons/sale_purchase/models/sale_order_line.py
+++ b/addons/sale_purchase/models/sale_order_line.py
@@ -201,7 +201,7 @@ class SaleOrderLine(models.Model):
         if line_description:
             name += line_description

-        return {
+        purchase_line_vals = {
             'name': name,
             'product_qty': purchase_qty_uom,
             'product_id': self.product_id.id,
@@ -212,6 +212,11 @@ class SaleOrderLine(models.Model):
             'order_id': purchase_order.id,
             'sale_line_id': self.id,
         }
+        if self.analytic_distribution:
+            purchase_line_vals['analytic_distribution'] = self.analytic_distribution
+        elif self.order_id.account_analytic_id:
+            purchase_line_vals['analytic_distribution'] = {self.order_id.account_analytic_id.id: 100.0}
+        return purchase_line_vals

     def _purchase_service_match_supplier(self, warning=True):
         # determine vendor of the order (take the first matching company and product)
```

ps: I did not test that code

Reply:
Hello @xavierbol 
I have done the changes.

Thank You :slightly_smiling_face: